src/test/groovy/org/sonarqube/gradle/SonarQubePluginTest.groovy

```diff
@@ -811,6 +811,47 @@ class SonarQubePluginTest extends Specification {
     properties["sonar.kotlin.gradleProjectRoot"] == rootProject.projectDir as String
   }
 
+  def "Add implicit compile dependencies by default"() {
+    def rootProject = ProjectBuilder.builder().withName("root").build()
+    rootProject.pluginManager.apply(JavaPlugin)
+
+    when:
+    rootProject.pluginManager.apply(SonarQubePlugin)
+
+    then:
+    def sonarTask = rootProject.tasks.sonar
+    sonarTask.getDependsOn().size() == 2
+    sonarTask.getMustRunAfter().getDependencies(sonarTask).size() == 1
```

Comment:
Since it should always be the same 3 tasks, should we be explicit in the tasks that are expecting to list?